In [60]:
pip install pandas

Note: you may need to restart the kernel to use updated packages.


In [61]:
#1. Download dataset and import it into Python.

import pandas as pd

df = pd.read_csv('https://raw.githubusercontent.com/roizherrerapilar-hub/Mini-project---Shark-Attacks/refs/heads/main/Shark_Attacks_Dataset.csv', sep=';')
pd.set_option('display.max_columns', None) #Con esta funcion vemos todas las columnas que tiene el archivo


C:\Users\pilir\AppData\Local\Temp\ipykernel_19576\4068600238.py:5: DtypeWarning: Columns (0: Date, 1: Year, 2: Type, 3: Country, 4: State, 5: Location, 6: Activity, 7: Name, 8: Sex, 9: Age, 10: Injury, 11: Fatal Y/N, 12: Time, 13: Species , 14: Source, 15: pdf, 16: href formula, 17: href, 18: Case Number, 19: Case Number.1, 20: Unnamed: 21, 21: Unnamed: 22) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv('https://raw.githubusercontent.com/roizherrerapilar-hub/Mini-project---Shark-Attacks/refs/heads/main/Shark_Attacks_Dataset.csv', sep=';')


In [62]:
#2. Visualize the Dataframe 

df.head() 

,Date,Year,Type,Country,State,Location,Activity,Name,Sex,Age,Injury,Fatal Y/N,Time,Species,Source,pdf,href formula,href,Case Number,Case Number.1,original order,Unnamed: 21,Unnamed: 22,Unnamed: 23,Unnamed: 24,Unnamed: 25,Unnamed: 26,Unnamed: 27,Unnamed: 28,Unnamed: 29,Unnamed: 30,Unnamed: 31,Unnamed: 32,Unnamed: 33,Unnamed: 34,Unnamed: 35,Unnamed: 36,Unnamed: 37,Unnamed: 38,Unnamed: 39,Unnamed: 40,Unnamed: 41,Unnamed: 42,Unnamed: 43,Unnamed: 44,Unnamed: 45,Unnamed: 46,Unnamed: 47,Unnamed: 48,Unnamed: 49,Unnamed: 50,Unnamed: 51,Unnamed: 52,Unnamed: 53,Unnamed: 54,Unnamed: 55,Unnamed: 56,Unnamed: 57,Unnamed: 58,Unnamed: 59,Unnamed: 60,Unnamed: 61,Unnamed: 62,Unnamed: 63,Unnamed: 64,Unnamed: 65,Unnamed: 66,Unnamed: 67,Unnamed: 68,Unnamed: 69,Unnamed: 70,Unnamed: 71,Unnamed: 72,Unnamed: 73,Unnamed: 74,Unnamed: 75,Unnamed: 76,Unnamed: 77,Unnamed: 78,Unnamed: 79,Unnamed: 80,Unnamed: 81,Unnamed: 82,Unnamed: 83,Unnamed: 84,Unnamed: 85,Unnamed: 86,Unnamed: 87,Unnamed: 88,Unnamed: 89,Unnamed: 90,Unnamed: 91,Unnamed: 92,Unnamed: 93,Unnamed: 94,Unnamed: 95,Unnamed: 96,Unnamed: 97,Unnamed: 98,Unnamed: 99,Unnamed: 100,Unnamed: 101,Unnamed: 102,Unnamed: 103,Unnamed: 104,Unnamed: 105,Unnamed: 106,Unnamed: 107,Unnamed: 108,Unnamed: 109,Unnamed: 110,Unnamed: 111,Unnamed: 112,Unnamed: 113,Unnamed: 114,Unnamed: 115,Unnamed: 116,Unnamed: 117,Unnamed: 118,Unnamed: 119,Unnamed: 120,Unnamed: 121,Unnamed: 122,Unnamed: 123,Unnamed: 124,Unnamed: 125,Unnamed: 126,Unnamed: 127,Unnamed: 128,Unnamed: 129,Unnamed: 130,Unnamed: 131,Unnamed: 132,Unnamed: 133,Unnamed: 134,Unnamed: 135,Unnamed: 136,Unnamed: 137,Unnamed: 138,Unnamed: 139,Unnamed: 140,Unnamed: 141,Unnamed: 142,Unnamed: 143,Unnamed: 144,Unnamed: 145,Unnamed: 146,Unnamed: 147,Unnamed: 148,Unnamed: 149,Unnamed: 150,Unnamed: 151,Unnamed: 152,Unnamed: 153,Unnamed: 154,Unnamed: 155,Unnamed: 156,Unnamed: 157,Unnamed: 158,Unnamed: 159,Unnamed: 160,Unnamed: 161,Unnamed: 162,Unnamed: 163,Unnamed: 164,Unnamed: 165,Unnamed: 166,Unnamed: 167,Unnamed: 168,Unnamed: 169,Unnamed: 170,Unnamed: 171,Unnamed: 172,Unnamed: 173,Unnamed: 174,Unnamed: 175,Unnamed: 176,Unnamed: 177,Unnamed: 178,Unnamed: 179,Unnamed: 180,Unnamed: 181,Unnamed: 182,Unnamed: 183,Unnamed: 184,Unnamed: 185,Unnamed: 186,Unnamed: 187,Unnamed: 188,Unnamed: 189,Unnamed: 190,Unnamed: 191,Unnamed: 192,Unnamed: 193,Unnamed: 194,Unnamed: 195,Unnamed: 196,Unnamed: 197,Unnamed: 198,Unnamed: 199,Unnamed: 200,Unnamed: 201,Unnamed: 202,Unnamed: 203,Unnamed: 204,Unnamed: 205,Unnamed: 206,Unnamed: 207,Unnamed: 208,Unnamed: 209,Unnamed: 210,Unnamed: 211,Unnamed: 212,Unnamed: 213,Unnamed: 214,Unnamed: 215,Unnamed: 216,Unnamed: 217,Unnamed: 218,Unnamed: 219,Unnamed: 220,Unnamed: 221,Unnamed: 222,Unnamed: 223,Unnamed: 224,Unnamed: 225,Unnamed: 226,Unnamed: 227,Unnamed: 228,Unnamed: 229,Unnamed: 230,Unnamed: 231,Unnamed: 232,Unnamed: 233,Unnamed: 234,Unnamed: 235,Unnamed: 236,Unnamed: 237,Unnamed: 238,Unnamed: 239,Unnamed: 240,Unnamed: 241,Unnamed: 242,Unnamed: 243,Unnamed: 244,Unnamed: 245,Unnamed: 246,Unnamed: 247,Unnamed: 248,Unnamed: 249,Unnamed: 250,Unnamed: 251,Unnamed: 252,Unnamed: 253,Unnamed: 254
0,14/04,2026,UNprovoked,Maldives,Gaafu Alif Atoll,Kooddoo,Swimming,Not stated - on honeymoon,M,?,Leg strpped off flesh later amputated in hospital,N,?,Unknown,The U.S. Sun: Simon De Marchi,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN

In [63]:
#3. Indicate the number of columns and rows. 
df.shape 

(39196, 255)

In [64]:
#4. Delete the columns with null values (from column 16: source)
df = df.drop(columns=df.columns[15:]) 
df.head()

,Date,Year,Type,Country,State,Location,Activity,Name,Sex,Age,Injury,Fatal Y/N,Time,Species,Source
0,14/04,2026,UNprovoked,Maldives,Gaafu Alif Atoll,Kooddoo,Swimming,Not stated - on honeymoon,M,?,Leg strpped off flesh later amputated in hospital,N,?,Unknown,The U.S. Sun: Simon De Marchi
1,3rd April,2026,Unprovoked,Australia,South Australia,Middleton Beach Fleurieu Peninsula Adelaide,Surfing,Oliver Tokic-Bensley,M,16,Bite wound to R ankle,N,?,Bronze Whaler,ABC News: The Guardian: Andrew Currie and Bob...
2,26th March,2026,Unprovoked,Bahamas,Andros Island,Fresh Creek,Swimming,Australian woman,F,22,Severe injuries to R arm,N,1730hrs,Unknown,WIC News: Melissa Michaelson
3,25th March,2026,Unprovoked,Australia,South Australia,Cape Jaffa Limestone Coast,Diving,Luke Kuhn,M,?,NaN,N,?,Great White Shark 3.5m (11.5ft),Bob Myatt GSAF
4,22nd-23rd March,2026,Unprovoked,Australia,NSW,Little Avalon Beach,Surfing,Unknown,M,30+,Graze injuries R leg and ankle,N,?,Unknown,Melissa Michaelson: Instagram


In [65]:
#5. Indicate the names of the columns. 
df.columns 

Index(['Date', 'Year', 'Type', 'Country', 'State', 'Location', 'Activity',
       'Name', 'Sex', 'Age', 'Injury', 'Fatal Y/N', 'Time', 'Species ',
       'Source'],
      dtype='str')

In [66]:
#6. Delete the rows with null values. 
df.dropna()
df.head()

,Date,Year,Type,Country,State,Location,Activity,Name,Sex,Age,Injury,Fatal Y/N,Time,Species,Source
0,14/04,2026,UNprovoked,Maldives,Gaafu Alif Atoll,Kooddoo,Swimming,Not stated - on honeymoon,M,?,Leg strpped off flesh later amputated in hospital,N,?,Unknown,The U.S. Sun: Simon De Marchi
1,3rd April,2026,Unprovoked,Australia,South Australia,Middleton Beach Fleurieu Peninsula Adelaide,Surfing,Oliver Tokic-Bensley,M,16,Bite wound to R ankle,N,?,Bronze Whaler,ABC News: The Guardian: Andrew Currie and Bob...
2,26th March,2026,Unprovoked,Bahamas,Andros Island,Fresh Creek,Swimming,Australian woman,F,22,Severe injuries to R arm,N,1730hrs,Unknown,WIC News: Melissa Michaelson
3,25th March,2026,Unprovoked,Australia,South Australia,Cape Jaffa Limestone Coast,Diving,Luke Kuhn,M,?,NaN,N,?,Great White Shark 3.5m (11.5ft),Bob Myatt GSAF
4,22nd-23rd March,2026,Unprovoked,Australia,NSW,Little Avalon Beach,Surfing,Unknown,M,30+,Graze injuries R leg and ankle,N,?,Unknown,Melissa Michaelson: Instagram


In [ ]:
#7. Basic overview of the dataset. 
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 39196 entries, 0 to 39195
Data columns (total 15 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   Date       7087 non-null   str   
 1   Year       7085 non-null   object
 2   Type       7069 non-null   str   
 3   Country    7037 non-null   str   
 4   State      6600 non-null   str   
 5   Location   6520 non-null   str   
 6   Activity   6504 non-null   str   
 7   Name       6869 non-null   str   
 8   Sex        6509 non-null   str   
 9   Age        4093 non-null   str   
 10  Injury     7051 non-null   str   
 11  Fatal Y/N  6526 non-null   str   
 12  Time       3560 non-null   str   
 13  Species    3956 non-null   str   
 14  Source     7067 non-null   str   
dtypes: object(1), str(14)
memory usage: 4.5+ MB


Comentarios:
- Vemos que el data type de fecha es string y deberia ser data type object. 
- Los valores no nulos de las columnas: date, year, type, country, injury, source tienen valores similares. 
- Vemos que las columnas de age y time presentan en algunas filas un signo de interrogacion por falta de datos, por lo que mas adelante analizaremos que ocurre. 


Clasificacion de las columnas: 
- Categorical: type, country, state, location, activity, name, sex, injury, fatal Y/N, species, source. 
- Numerical: date, year, age, time. 

In [ ]:
#8. Indicate the names of the columns.
print(df.columns.tolist())  


['Date', 'Year', 'Type', 'Country', 'State', 'Location', 'Activity', 'Name', 'Sex', 'Age', 'Injury', 'Fatal Y/N', 'Time', 'Species ', 'Source']


In [ ]:
#9. Analysis of categorical variables. 
df.describe()

,Date,Year,Type,Country,State,Location,Activity,Name,Sex,Age,Injury,Fatal Y/N,Time,Species,Source
count,7087,7085,7069,7037,6600,6520,6504,6869,6509,4093,7051,6526,3560,3956,7067
unique,6096,264,14,253,948,4631,1612,5801,11,178,4197,12,477,1746,5413
top,1957,2017,Unprovoked,USA,Florida,"New Smyrna Beach, Volusia County",Surfing,male,M,17,FATAL,N,Afternoon,White shark,"K. McMurray, TrackingSharks.com"
freq,11,141,5237,2580,1192,191,1151,679,5682,174,863,4943,215,194,131


Comentarios: 
- Vemos que la columna tiene presenta solo 14 valores unicos provocados siendo el valor total de 5413. Lo cual podemos deducir que hay algo raro que debemos de analizar posteriormente. 
- Vemos que la columna de country tiene 253 valores unicos, siendo el mas dominante, USA. Por lo tanto, se deduce que los ataques se concentran en USA. 
- Nos resulta sospechoso ver que el valor mas frecuente en la columna de date sea 1957, dado que no tiene formato de fecha mm/yy/dd. 
- En la columna de especies, nos llama la atencion que siendo 1746 especies unicas White Shark es la especie con mas ataques registrados en las costas de Florida. 